common/djangoapps/third_party_auth/models.py

```diff
@@ -241,22 +241,31 @@ def get_register_form_data(cls, pipeline_kwargs):
             values for that field. Where there is no value, the empty string
             must be used.
         """
+        current_provider = kwargs.pop('current_provider', None)
+        registration_form_data = {}
+
         # Details about the user sent back from the provider.
         details = pipeline_kwargs.get('details')
+        registration_form_data['email'] = details.get('email', '')
+        registration_form_data['name'] = details.get('fullname', '')
+
+        # There are often nonstandard requirements for some registration flows,
+        # like the user's country code.
+        if current_provider.required_extra_attributes:
+            attributes = pipeline_kwargs['response']['attributes']
+            for attr, val in current_provider.required_extra_attributes.iteritems():
+                if val:
+                    registration_form_data[attr] = attributes[val]
```

Comment:
I'm not convinced that it adds much value to make this `required_extra_attributes` defined on the provider... Would there be any downside to simply writing

```python
registration_form_data['country'] = details.get('country', '')
```
unconditionally?

Or if it is conditional, I feel like it should be conditional on whether or not the Open edX instance has the "country" field enabled for this form - not whether or not the provider says it sometimes "requires" the extra country attribute.

Put another way: I can't think of a reason why we wouldn't want to pre-fill the country field if it's enabled and if the data is available from the provider, so I don't think we need to make that behavior optional at any level.